src/priorityqueue.jl

```diff
@@ -48,26 +47,121 @@ struct PriorityQueue{K,V,O<:Ordering} <: AbstractDict{K,V}
         pq = new{K,V,O}(xs, o, index)
 
         # heapify
-        for i in heapparent(length(pq.xs)):-1:1
+        for i in heapparent(length(get_xs(pq))):-1:1
             percolate_down!(pq, i)
         end
 
         pq
     end
 end
 
+"""
+    IntPriorityQueue{K, V}([ord])
+
+PriorityQueue that accepts integer in 1:n as keys.
+"""
+struct IntPriorityQueue{K,V,O<:Ordering} <: AbstractPriorityQueue{K,V,O}
```

Comment:
I feel like:

```suggestion
struct IntPriorityQueue{V,O<:Ordering} <: AbstractPriorityQueue{Int,V,O}
```

Reply:
That's a good point since keys must be in `1:n`.
A better way might be to enforce it might be `K<:Integer`.

Or `K<:UnsignedInteger`